src/components/PostDetails.tsx

```diff
@@ -1,126 +1,56 @@
 import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  commentsFetch,
+  commentsRemove,
+  commentsRemoveFast,
+} from '../features/commentsSlice';
+import { Status } from '../types/Status';
+
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const { comments, status } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
-
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
+  const deleteComment = async (commentID: number) => {
+    dispatch(commentsRemoveFast(commentID));
```

Comment:
it's not good naming `commentsRemove` and `commentsRemoveFast`
better to rename `commentsRemoveFast` to `setComments` as it will make a code cleaner

Reply:
Done